### PR TITLE
typescript-jquery: Update to work with Typescript strict mode

### PR DESCRIPTION
**Pull request by janjongboom**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @nicokoenig @topce @akehir

### Description of the PR

When compiling an API generated using the typescript-jquery generator against Typescript 3.6.2 in strict mode there are a number of issues with the generated code.

* `defaultExtraJQueryAjaxSettings` is of type `JQueryAjaxSettings | undefined` but is set to `null` (valid in normal mode, not valid in strict mode).
* In `extendObj` `hasOwnProperty` is called on `objB`. But `objB` is not guaranteed to be an object as it inherits from unknown type `T1`.
* Properties in the `Configuration` class are marked as `string`, but have no initializer, thus giving false safety when consuming this class (values could be undefined too).

This PR adds some small guards around these issues.

### CODE REVIEW

File: samples/client/petstore/typescript-jquery/default/api/PetApi.ts [hunk 1]
```diff
@@ -49,19 +49,19 @@ export class PetApi {
     /**
      * 
      * @summary Add a new pet to the store
-     * @param body Pet object that needs to be added to the store
+     * @param pet Pet object that needs to be added to the store
      */
-    public addPet(body: models.Pet, extraJQueryAjaxSettings?: JQueryAjaxSettings): JQuery.Promise<
+    public addPet(pet: models.Pet, extraJQueryAjaxSettings?: JQueryAjaxSettings): JQuery.Promise<
```
Comment (macjohnny) [inline samples/client/petstore/typescript-jquery/default/api/PetApi.ts]: @janjongboom please run the https://github.com/OpenAPITools/openapi-generator/blob/master/bin/typescript-jquery-petstore-all.sh script, not the openapi3 version.

File: samples/client/petstore/typescript-jquery/default/api/PetApi.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export class PetApi {
 
     private extendObj<T1, T2 extends T1>(objA: T2, objB: T2): T1|T2 {
         for (let key in objB) {
-            if (objB.hasOwnProperty(key)) {
+            if (objB[key]) {
```
Comment (macjohnny) [inline samples/client/petstore/typescript-jquery/default/api/PetApi.ts]: where does this change come from?

Comment (janjongboom) [reply samples/client/petstore/typescript-jquery/default/api/PetApi.ts]: Hmm... That should not be there, something went wrong with rebasing. Will repush.

File: samples/client/petstore/typescript-jquery/default/api/PetApi.ts (hunk 1 above)
Comment (janjongboom) [reply samples/client/petstore/typescript-jquery/default/api/PetApi.ts]: Pushed.

**Comment by janjongboom** [comment]
@macjohnny Pushed changes as requested.

File: samples/client/petstore/typescript-jquery/default/model/InlineObject1.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import * as models from './models';
+
+export interface InlineObject1 {
```
[the remark this answers is not in the document]
Comment (janjongboom) [reply samples/client/petstore/typescript-jquery/default/model/InlineObject1.ts]: @macjohnny Removed and re-pushed. However, these were generated by the tool, so interesting to see where they'd come from.

[the remark this answers is not in the document]
Comment (macjohnny) [reply samples/client/petstore/typescript-jquery/default/model/InlineObject1.ts]: these files are generated with the `bin/openapi3/typescript-fetch-petstore-all.sh` command. the output is supposed to be the same as with `bin/typescript-fetch-petstore-all.sh`, but it isn't, so it needs to be addressed in a separate issue.

[the remark this answers is not in the document]
Comment (macjohnny) [reply samples/client/petstore/typescript-jquery/default/model/InlineObject1.ts]: CI currently checks against the output of `bin/typescript-fetch-petstore-all.sh`

**Comment by janjongboom** [comment]
Rebased and pushed to restart tests.

File: samples/client/petstore/typescript-jquery/npm/model/InlineObject1.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
```
Comment (macjohnny) [inline samples/client/petstore/typescript-jquery/npm/model/InlineObject1.ts]: please delete this file

File: samples/client/petstore/typescript-node/default/api/petApi.ts [hunk 5]
```diff
@@ -83,17 +83,17 @@ export class PetApi {
     /**
      * 
      * @summary Add a new pet to the store
-     * @param body Pet object that needs to be added to the store
+     * @param pet Pet object that needs to be added to the store
      */
-    public async addPet (body: Pet, options: {headers: {[name: string]: string}} = {headers: {}}) : Promise<{ response: http.ClientResponse; body?: any;  }> {
```
Comment (macjohnny) [inline samples/client/petstore/typescript-node/default/api/petApi.ts]: this is unrelated

File: samples/client/petstore/typescript-jquery/npm/model/InlineObject1.ts (hunk 4 above)
Comment (janjongboom) [reply samples/client/petstore/typescript-jquery/npm/model/InlineObject1.ts]: @macjohnny I'm not sure where these come from, but the file is referenced in https://github.com/OpenAPITools/openapi-generator/pull/3969/files/c99fcd7b5936800b4ef0d38e92ebd27dc86fadcb#diff-969fc1a4f92fd269b7aac842f2049bb7R15 as well. This is just the output of the build step.

File: samples/client/petstore/typescript-node/default/api/petApi.ts (hunk 5 above)
Comment (janjongboom) [reply samples/client/petstore/typescript-node/default/api/petApi.ts]: Similar to above. Apparently something was landed earlier but the samples weren't updated (generated by the build script). I'm happy to revert this file back to master but shouldn't samples always be reflective of the latest version of the tool?

Comment (macjohnny) [reply samples/client/petstore/typescript-node/default/api/petApi.ts]: I will update your branch with the correct samples

**Comment by macjohnny** [comment]
@janjongboom I cleaned the samples in your branch. By the way: no need to re-base, PRs are squashed.

**Comment by janjongboom** [comment]
@macjohnny OK, thanks!

**Comment by wing328** [comment]
@janjongboom thanks for the PR, which has been included in v4.2.0 release: https://twitter.com/oas_generator/status/1189824932345069569
